kevm-pyk/src/kevm_pyk/kproj/evm-semantics/word.md

```diff
@@ -585,6 +585,81 @@ Range of types
  // ------------------------------------------
     rule N <<Byte M => N <<Int (8 *Int M)
     rule N >>Byte M => N >>Int (8 *Int M)
+```
+
+Auxiliary functions for slot-update reasoning
+---------------------------------------------
+
+Slot updates are performed by the compiler with the help of masks, which are words of the form `1111-ONES-111100000-WIDTH-00000011111-SHIFT-11111`.
+The following auxiliary functions support the associated automated reasoning.
+
+-   `#getFirstOneBit` returns the 0-based index of the first non-zero bit of a word
+
+```k
+    syntax Int ::=  #getFirstOneBit(Int) [function, total]
+
+    rule [gfo-succ]: #getFirstOneBit(X:Int) => log2Int ( X &Int ( ( maxUInt256 xorInt X ) +Int 1 ) ) requires #rangeUInt(256, X) andBool X =/=Int 0 [preserves-definedness]
+    rule [gfo-fail]: #getFirstOneBit(_:Int) => -1                                                    [owise]
+```
```

Comment:
I'm not sure why these definitions are moved to this file. Do we need them in the LLVM backend sometimes? Maybe when the booster is sending terms to the LLVM backend?

Reply:
Will explain in PR info shortly.